tm2/pkg/store/types/gas.go

```diff
@@ -79,7 +79,7 @@ func (g *basicGasMeter) Limit() Gas {
 }
 
 func (g *basicGasMeter) Remaining() Gas {
-	return g.Limit() - g.GasConsumedToLimit()
+	return overflow.Sub64p(g.Limit(), g.GasConsumedToLimit())
```

Comment:
Nitpick: Looking further down the code, I noticed that in a similar case, panics are wrapped with an error. 
```go
	consumed, ok := overflow.Add64(g.consumed, amount)
	if !ok {
		panic(GasOverflowError{descriptor})
	}
```

Maybe it'd be better to do the same thing for consistency? What do you think?

Reply:
In the case you mention, a user descriptor is known and justifies a custom panic message. In `Remaining()`, there is no specific context so we used the generic panic message.